### ISSUE TITLE
Missing Packages

### ISSUE THREAD

**Issue by audrummer15** [linnovate/mean#1597]
I am on Ubuntu 16.04
NodeJS v6.2.2 NPM 3.9.5
MEAN Status: Worker Died

After using mean-cli to init a project, I had to install the following manually with npm:

- cookie-parser
- express-validator
- method-override
- passport

**Comment by obcidio** [comment]
Similar issue - on init, getting this error:
`mv: cannot stat `loanApp/packages/custom/meanStarter': No such file or directory`

After init, there is no 'packages' directory.

**Comment by liorkesos** [comment]
Are you sure the project was cloned successfully ? Please paste the output
immidiattly after you ran mean init

On Tue, Jun 21, 2016 at 7:16 AM, James Lopez <notifications@github.com>
wrote:

> Similar issue - on init, getting this error:
> mv: cannot statloanApp/packages/custom/meanStarter': No such file or
> directory`
>
> After init, there is no 'packages' directory.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/linnovate/mean/issues/1597#issuecomment-227342661>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAO9Iw4DL1NBqAUYQ8SPkz6LHE_fXUyBks5qN3OdgaJpZM4I5nvO>
> .
>



-- 

Lior Kesos - http://www.linnovate.net
Linnovate - Community Infrastructure Care
mail: lior@linnovate.net
office: +972 722500881
cell: +972 524305252
skype: liorkesos

**Comment by gtrrz-victor** [comment]
I have the same problem. It can see on commit https://github.com/linnovate/mean/commit/eb916b0b0da517cb2ede60268a3d729e98048878 the folder _packages_ had been removed. It may be the cause.

**Comment by obcidio** [comment]
`
web01:~/meanio$ mean init myApp
? What would you name your mean app? myApp
Cloning branch: master into destination folder: myApp
git clone  --depth 1  -bmaster https://github.com/linnovate/mean.git "myApp"
Cloning into 'myApp'...

  Added the "remote" upstream origin
mv: cannot stat `myApp/packages/custom/meanStarter': No such file or directory

Do you want to set up an admin user?
? Yes N

#############################################

All Done! - Now lets install the dependencies...

  Install node package dependencies:
    $ cd myApp && npm install
  Bower install should be triggered for client side dependencies.
  If it did not run invoke it manually...
    $ cd myApp && bower install
  Run the app by running:
    $ cd myApp and then run..
    $ gulp
`

**Comment by obcidio** [comment]
Executed init using `mean init -b 0.6.0` and it seemed to work as expected.

**Comment by audrummer15** [comment]
```
ab@ubuntu:~/Development/web$ mean init which-lineup-linnovate
? What would you name your mean app? which-lineup-linnovate
Cloning branch: master into destination folder: which-lineup-linnovate
git clone  --depth 1  -bmaster https://github.com/linnovate/mean.git "which-lineup-linnovate"
Cloning into 'which-lineup-linnovate'...

  Added the "remote" upstream origin
mv: cannot stat 'which-lineup-linnovate/packages/custom/meanStarter': No such file or directory

Do you want to set up an admin user?
? Yes y
? Please provide your email so we can create your first admin user xxx@yyy.com


/usr/local/lib/node_modules/mean-cli/node_modules/rx/dist/rx.js:77
    throw e;
    ^
true
ab@ubuntu:~/Development/web$
```

I think this may be two separate issues. Creating an admin user also seems to fail. Here is the output without admin creation

```
ab@ubuntu:~/Development/web$ mean init which-lineup-linnovate
? What would you name your mean app? which-lineup-linnovate
Cloning branch: master into destination folder: which-lineup-linnovate
git clone  --depth 1  -bmaster https://github.com/linnovate/mean.git "which-lineup-linnovate"
Cloning into 'which-lineup-linnovate'...

  Added the "remote" upstream origin
mv: cannot stat 'which-lineup-linnovate/packages/custom/meanStarter': No such file or directory

Do you want to set up an admin user?
? Yes n

#############################################

All Done! - Now lets install the dependencies...

  Install node package dependencies:
    $ cd which-lineup-linnovate && npm install
  Bower install should be triggered for client side dependencies.
  If it did not run invoke it manually...
    $ cd which-lineup-linnovate && bower install
  Run the app by running:
    $ cd which-lineup-linnovate and then run..
    $ gulp

ab@ubuntu:~/Development/web$ 
```

In any case, after running npm install, it does just fine. Then gulp produces this:

```
ab@ubuntu:~/Development/web/which-lineup-linnovate$ gulp
(node:37391) fs: re-evaluating native module sources is not supported. If you are using the graceful-fs module, please update it to a more recent version.
Invoking gulp - development
[13:28:19] Using gulpfile ~/Development/web/which-lineup-linnovate/gulpfile.js
[13:28:19] Starting 'clean'...
[13:28:19] Finished 'clean' after 5.51 ms
[13:28:19] Starting 'default'...
[13:28:19] Starting 'webpack:build-dev'...
[13:28:19] Starting 'clean'...
[13:28:19] Starting 'less'...
[13:28:19] Starting 'sass'...
[13:28:19] Starting 'csslint'...
[13:28:20] Starting 'env:development'...
[13:28:20] Finished 'env:development' after 16 μs
[13:28:20] Starting 'devServe'...
[13:28:20] Finished 'devServe' after 29 ms
[13:28:20] Starting 'watch'...
[13:28:20] Finished 'watch' after 62 ms
[13:28:20] Finished 'clean' after 260 ms
[13:28:20] Finished 'less' after 264 ms
[13:28:20] Finished 'sass' after 157 ms
[13:28:20] csslint: 0 files lint free
[13:28:20] Finished 'csslint' after 149 ms
[13:28:20] [nodemon] 1.9.2
[13:28:20] [nodemon] to restart at any time, enter `rs`
[13:28:20] [nodemon] watching: *.*
[13:28:20] [nodemon] starting `node --debug server.js`
Debugger listening on port 5858
Mean app started on port 3000 (development) cluster.worker.id: 0
[13:28:22] index.html reloaded.
[13:28:25] [nodemon] restarting due to changes...
[13:28:25] [webpack:build-dev] Hash: 1a6b118808961849a5bd
Version: webpack 1.13.1
Time: 5545ms
     Asset     Size  Chunks             Chunk Names
    app.js  2.76 MB       0  [emitted]  main
app.js.map   3.2 MB       0  [emitted]  main
chunk    {0} app.js, app.js.map (main) 2.69 MB [rendered]
    [0] ./app.js 1.66 kB {0} [built] [1 error]
    [1] ./bower_components/angular/angular-csp.css 937 bytes {0} [built]
    [2] ./~/css-loader!./bower_components/angular/angular-csp.css 527 bytes {0} [built]
    [3] ./~/css-loader/lib/css-base.js 1.51 kB {0} [built]
    [4] ./~/style-loader/addStyles.js 7.15 kB {0} [built]
    [5] ./~/angular-ui-select/select.min.css 870 bytes {0} [built]
    [6] ./~/css-loader!./~/angular-ui-select/select.min.css 4.34 kB {0} [built]
    [7] ./bower_components/bootstrap/dist/css/bootstrap.min.css 967 bytes {0} [built]
    [8] ./~/css-loader!./bower_components/bootstrap/dist/css/bootstrap.min.css 122 kB {0} [built]
    [9] ./bower_components/bootstrap/dist/fonts/glyphicons-halflings-regular.eot 26.9 kB {0} [built]
   [10] ./bower_components/bootstrap/dist/fonts/glyphicons-halflings-regular.woff2 24.1 kB {0} [built]
   [11] ./bower_components/bootstrap/dist/fonts/glyphicons-halflings-regular.woff 31.3 kB {0} [built]
   [12] ./bower_components/bootstrap/dist/fonts/glyphicons-halflings-regular.ttf 60.6 kB {0} [built]
   [13] ./bower_components/bootstrap/dist/fonts/glyphicons-halflings-regular.svg 145 kB {0} [built]
   [14] ./~/jquery/dist/jquery.js 263 kB {0} [built]
   [15] ./bower_components/angular/index.js 48 bytes {0} [built]
   [16] ./bower_components/angular/angular.js 1.17 MB {0} [built]
   [17] ./~/angular-ui-select/select.js 76.4 kB {0} [built]
   [18] ./bower_components/angular-mocks/angular-mocks.js 108 kB {0} [built]
   [19] ./bower_components/angular-cookies/index.js 60 bytes {0} [built]
   [20] ./bower_components/angular-cookies/angular-cookies.js 9.74 kB {0} [built]
   [21] ./bower_components/angular-resource/index.js 62 bytes {0} [built]
   [22] ./bower_components/angular-resource/angular-resource.js 34.9 kB {0} [built]
   [23] ./bower_components/angular-sanitize/index.js 62 bytes {0} [built]
   [24] ./bower_components/angular-sanitize/angular-sanitize.js 25.8 kB {0} [built]
   [25] ./~/angular-ui-router/release/angular-ui-router.js 170 kB {0} [built]
   [26] ./~/angular-jwt/index.js 67 bytes {0} [built]
   [27] ./~/angular-jwt/dist/angular-jwt.js 3.63 kB {0} [built]
   [28] ./bower_components/angular-bootstrap/ui-bootstrap-tpls.js 261 kB {0} [built]
   [29] ./node_modules \/meanio-(admin|system|users|circles)\/public\/(?!tests|assets|views)(.*)\.js$ 1.39 kB {0} [built]
   [30] ./~/meanio-admin/public/controllers/admin.js 1.08 kB {0} [optional] [built]
   [31] ./~/meanio-admin/public/controllers/example.js 545 bytes {0} [optional] [built]
   [32] ./~/meanio-admin/public/controllers/modules.js 413 bytes {0} [optional] [built]
   [33] ./~/meanio-admin/public/controllers/settings.js 3.07 kB {0} [optional] [built]
   [34] ./~/meanio-admin/public/controllers/themes.js 1.63 kB {0} [optional] [built]
   [35] ./~/meanio-admin/public/controllers/users.js 3.13 kB {0} [optional] [built]
   [36] ./~/meanio-admin/public/directives/editable.js 3.34 kB {0} [optional] [built]
   [37] ./~/meanio-admin/public/index.js 119 bytes {0} [optional] [built]
   [38] ./~/meanio-admin/public/assets/css/admin.css 891 bytes {0} [built]
   [39] ./~/css-loader!./~/meanio-admin/public/assets/css/admin.css 3.04 kB {0} [built]
   [40] ./~/zeroclipboard/dist/ZeroClipboard.js 85.6 kB {0} [built]
   [41] ./~/ng-clip/dest/ng-clip.min.js 1.18 kB {0} [built]
   [42] ./~/meanio-admin/public/routes/admin.js 1.22 kB {0} [optional] [built]
   [43] ./~/meanio-admin/public/services/module-settings.js 992 bytes {0} [optional] [built]
   [44] ./~/meanio-admin/public/services/modules.js 341 bytes {0} [optional] [built]
   [45] ./~/meanio-admin/public/services/settings.js 1.01 kB {0} [optional] [built]
   [46] ./~/meanio-admin/public/services/users.js 314 bytes {0} [optional] [built]
   [47] ./~/meanio-circles/public/controllers/circles.js 736 bytes {0} [optional] [built]
   [48] ./~/meanio-circles/public/index.js 51 bytes {0} [optional] [built]
   [49] ./~/meanio-circles/public/assets/css/circles.css 897 bytes {0} [built]
   [50] ./~/css-loader!./~/meanio-circles/public/assets/css/circles.css 524 bytes {0} [built]
   [51] ./~/meanio-circles/public/routes/circles.js 1.6 kB {0} [optional] [built]
   [52] ./~/meanio-circles/public/services/circles.js 491 bytes {0} [optional] [built]
   [53] ./~/meanio-system/public/routes/system.js 1.23 kB {0} [optional] [built]
   [54] ./~/meanio-system/public/services/config.js 554 bytes {0} [optional] [built]
   [55] ./~/meanio-system/public/services/global.js 3.82 kB {0} [optional] [built]
   [56] ./~/meanio-system/public/services/interceptor.js 881 bytes {0} [optional] [built]
   [57] ./~/meanio-system/public/services/menus.js 218 bytes {0} [optional] [built]
   [58] ./~/meanio-system/public/system.js 522 bytes {0} [optional] [built]
   [59] ./~/meanio-users/public/controllers/meanUser.js 4.07 kB {0} [optional] [built]
   [60] ./~/meanio-users/public/index.js 38 bytes {0} [optional] [built]
   [61] ./~/meanio-users/public/routes/auth.js 346 bytes {0} [optional] [built]
   [62] ./~/meanio-users/public/services/meanUser.js 6.58 kB {0} [optional] [built]

ERROR in ./app.js
Module not found: Error: Cannot resolve 'file' or 'directory' ./packages in /home/adambrown/Development/web/which-lineup-linnovate
 @ ./app.js 54:14-94
[13:28:25] Finished 'webpack:build-dev' after 5.55 s
[13:28:25] Starting 'development'...
[13:28:25] Finished 'development' after 1.89 μs
[13:28:25] [nodemon] starting `node --debug server.js`
Debugger listening on port 5858
Mean app started on port 3000 (development) cluster.worker.id: 0
[13:28:27] index.html reloaded.

```

**Comment by audrummer15** [comment]
When doing as obcidio suggests, I get an error connecting to mongo. However, `mean status` tells me that the DB connection is successful. See below for outputs:
```
ab@ubuntu:~/Development/web/which-lineup-linnovate$ mean status

    MEAN Status
    -----------

    MEAN VERSION: 0.6.1

    DB connection successful!

    MongoDB URI: mongodb://localhost/mean-dev

    checking meanio and global mean-cli versions
    mean-cli at latest version: 0.12.10
    meanio at latest version: 0.9.1
ab@ubuntu:~/Development/web/which-lineup-linnovate$ 
```

```
ab@ubuntu:~/Development/web/which-lineup-linnovate$ gulp
(node:38013) fs: re-evaluating native module sources is not supported. If you are using the graceful-fs module, please update it to a more recent version.
Invoking gulp - development
[13:33:52] Using gulpfile ~/Development/web/which-lineup-linnovate/gulpfile.js
[13:33:52] Starting 'clean'...
[13:33:52] Finished 'clean' after 5.54 ms
[13:33:52] Starting 'default'...
[13:33:52] Starting 'webpack:build-dev'...
[13:33:52] Starting 'clean'...
[13:33:52] Starting 'less'...
[13:33:52] Starting 'sass'...
[13:33:52] Starting 'csslint'...
[13:33:52] Starting 'env:development'...
[13:33:52] Finished 'env:development' after 19 μs
[13:33:52] Starting 'devServe'...
[13:33:53] Finished 'devServe' after 33 ms
[13:33:53] Starting 'watch'...
[13:33:53] Finished 'watch' after 297 ms
[13:33:53] Finished 'clean' after 507 ms
[13:33:53] [nodemon] 1.9.2
[13:33:53] [nodemon] to restart at any time, enter `rs`
[13:33:53] [nodemon] watching: *.*
[13:33:53] [nodemon] starting `node --debug server.js`
Debugger listening on port 5858
WARNING: No configurations found in configuration directory:/home/adambrown/Development/web/which-lineup-linnovate/config
WARNING: To disable this warning set SUPPRESS_NO_CONFIG_WARNING in the environment.
[13:33:53] Finished 'less' after 808 ms
[13:33:53] Finished 'sass' after 702 ms
Error: Invalid mongodb uri. Missing hostname
**Could not connect to MongoDB. Please ensure mongod is running and restart MEAN app.**
[13:33:53] [nodemon] clean exit - waiting for changes before restart
[13:33:55] csslint: 3 files lint free
[13:33:55] Finished 'csslint' after 2.5 s
```

**Comment by obcidio** [comment]
Using the branch method, I was not able to get a stable instance up and running. Here's what I did to get a stable instance running:

1. Checkout the 0.6.0 branch and copied the packages directory to another location
2. Ran `mean init` as usual (no branch specification)
3. Copied the packages directory into the new directory created by the init script
4. Ran `npm install --verbose` (running it without verbose never produced a stable instance for some reason)

These steps produced a stable running instance

Also note, because of my server setup, I had to add a rule for the loopback interface in my iptables.

**Comment by itshakeel23** [comment]
Hi

I am facing problem to setup new application. I followed all the above steps but still the page doesn't load "http://localhost:3000" and on console it shows:

Invoking gulp - development
[14:51:06] Using gulpfile ~/projects/meanio/myApp/gulpfile.js
[14:51:06] Starting 'clean'...
[14:51:06] Finished 'clean' after 31 ms
[14:51:06] Starting 'default'...
[14:51:06] Starting 'webpack:build-dev'...
[14:51:06] Starting 'clean'...
[14:51:06] Starting 'less'...
[14:51:10] Starting 'sass'...
[14:51:11] Starting 'csslint'...
[14:51:12] Starting 'env:development'...
[14:51:12] Finished 'env:development' after 40 μs
[14:51:12] Starting 'devServe'...
[14:51:16] Finished 'devServe' after 3.7 s
[14:51:16] Starting 'watch'...
[14:51:18] Finished 'watch' after 2.57 s
[14:51:18] Finished 'clean' after 12 s
[14:51:18] [nodemon] 1.9.2
[14:51:18] [nodemon] to restart at any time, enter `rs`
[14:51:18] [nodemon] watching: *.*
[14:51:18] [nodemon] starting `node --debug server.js`
Debugger listening on port 5858
[14:51:19] Finished 'less' after 13 s
[14:51:19] Finished 'sass' after 8.52 s
Mean app started on port 3000 (development) cluster.worker.id: 0
[14:53:04] csslint: 3 files lint free
[14:53:04] Finished 'csslint' after 1.87 min
[14:53:04] index.html reloaded.
GET / - - - - ms

Steps followed:
- cd ~/projects/meanio && git clone -b 0.6.0 https://github.com/linnovate/mean.git mean-060
- mean init myApp && cd myApp && cp -R ../mean-060/packages . (copied packages directory)
- npm install --verbose
- gulp

After that page keep loading without showing anything.

**Comment by lionelyoung** [comment]
@obcidio instructions worked for my on Mac OSX El Capitan

```
> node --version
v6.2.2
> npm --version
3.10.3
```

1. Clone the 0.6.0 branch, because master branch is missing the 'packages' directory:
  - ```git clone -b 0.6.0 https://github.com/linnovate/mean.git mean-060```
2. Initialize mean.io and don't create admin (answer N)
  - ```mean init myApp```
3. Copy over the packages directory
  - ```cp -R mean-060/packages myApp```
4. Run npm with verbose flag
  - ```cd myApp && npm install --verbose```
6. ```gulp```
7. http://localhost:3000

I clicked "Join" and made an account.  Seems to work so far.

**Comment by itshakeel23** [comment]
So it worked now. The problem was due to node version. I was using node 4.4.6. Upgraded to 6.2.2 and worked perfectly.

Thanks

**Comment by itshakeel23** [comment]
Why we don't have core packages in master branch ? I want to extend articles package but didn't find in packages.

**Comment by timelf123** [comment]
The core packages are in `node_modules` now under `meanio-*`

**Comment by rivkabenshalom** [comment]
fixed in 0.6.0
